### PR TITLE
Filtered prefixes by `subject_prefixes`

### PR DESCRIPTION
**Pull request by hrshdhgd**
- [x] All lexical matches generated are filtered using  `subject_prefixes` in `mondo.sssom.config.yml` [issue #49 ]
- [ ] issue #50 
- [x] issue #51

### CODE REVIEW

File: src/scripts/match-mondo-sources-all-lexical.py [hunk 1]
```diff
@@ -85,11 +102,105 @@ def run(input:str, config: str, rules:str, output: str):
     
     # msdf.prefix_map = sssom_yaml['curie_map']
     # msdf.metadata = sssom_yaml['global_metadata']
+    msdf.df = filter_prefixes(df=msdf.df, filter_prefixes=prefix_of_interest, features=[SUBJECT_ID, OBJECT_ID])
+    # TODO: replace line below by msdf.remove_mappings(mapping_msdf) once imported from SSSOM.
```
Comment (matentzn) [inline src/scripts/match-mondo-sources-all-lexical.py:106]: Shall we make an sssom release and do this, or do you think its not worth waiting?

File: src/scripts/match-mondo-sources-all-lexical.py [hunk 2]
```diff
@@ -85,11 +102,105 @@ def run(input:str, config: str, rules:str, output: str):
...
         write_table(msdf, f)
 
+    objects = msdf.df[OBJECT_ID].drop_duplicates()
+    prefixes = objects.str.split(":").str.get(0).drop_duplicates()
+    prefix_args = tuple([x+":%" for _, x in prefixes.iteritems() if x != "MONDO"])
+    kwargs = {"subject_id": ("MONDO:%",), "object_id": prefix_args}
+    with open(str(SRC / Path(output.replace("lexical", "lexical-2"))), "w") as f:
+        filter_file(
+            input=str(SRC / Path(output)),
+            output=f,
```
Comment (matentzn) [inline src/scripts/match-mondo-sources-all-lexical.py:133]: I have never seen this way of doing it but looks cool. Learned something new!

File: src/scripts/match-mondo-sources-all-lexical.py [hunk 3]
```diff
@@ -85,11 +102,105 @@ def run(input:str, config: str, rules:str, output: str):
...
+            **kwargs
+            )
+
+# TODO: Replace the method below by sssom.utils.filter_prefixes.
```
Comment (matentzn) [inline src/scripts/match-mondo-sources-all-lexical.py:137]: Maybe if there are these two case (and the remove_mappings one above), lets keep the PR open and make an sssom release to fix this. Just to be clean, we are not in a super rush now.

**Review by matentzn — CHANGES_REQUESTED** [review]
Just changing to using sssom-py instead of the duplicated methods

File: src/scripts/match-mondo-sources-all-lexical.py (hunk 1 above)
Comment (hrshdhgd) [reply src/scripts/match-mondo-sources-all-lexical.py:106]: I think we should make a release.

File: src/scripts/match-mondo-sources-all-lexical.py (hunk 3 above)
Comment (hrshdhgd) [reply src/scripts/match-mondo-sources-all-lexical.py:137]: I agree!

**Pull request merged**